### PR TITLE
Add Sheep Shearing by punch #139

### PR DESCRIPTION
**Pull request by SFort**
I don't know if this is the exact behaviour it used to be
The current code requires the damage to be done by a Player
(is all damage supposed to shear?)

### CODE REVIEW

File: src/main/java/com/unascribed/fabrication/mixin/i_woina/old_sheep_shear/MixinSheepEntity.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	@Shadow
+	public abstract void setSheared(boolean sheared);
+	@Shadow
+	public abstract DyeColor getColor();
+
+	protected MixinSheepEntity(EntityType<? extends AnimalEntity> entityType, World world) {
+		super(entityType, world);
+	}
+
+	public boolean damage(DamageSource source, float amount) {
```
Comment (exaskye) [inline src/main/java/com/unascribed/fabrication/mixin/i_woina/old_sheep_shear/MixinSheepEntity.java]: I don't like the override here; generally I prefer to inject at the lowest class in the hierarchy that contains an override and use an `instanceof` check for less conflict potential.

File: src/main/java/com/unascribed/fabrication/mixin/i_woina/old_sheep_shear/MixinSheepEntity.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+			ItemEntity itemEntity = this.dropStack(wool, 1);
+			if (itemEntity != null) {
+				itemEntity.setVelocity(itemEntity.getVelocity().add((this.random.nextFloat() - this.random.nextFloat()) * 0.1F, this.random.nextFloat() * 0.05F, (this.random.nextFloat() - this.random.nextFloat()) * 0.1F));
+			}
+		}
+		return super.damage(source, amount);
+	}
+
+	@Surrogate
+	public PassiveEntity createChild(ServerWorld world, PassiveEntity entity) { return null; }
```
Comment (exaskye) [inline src/main/java/com/unascribed/fabrication/mixin/i_woina/old_sheep_shear/MixinSheepEntity.java]: What is this surrogate method for?

**Comment by exaskye** [comment]
> The current code requires the damage to be done by a Player

This is correct, to the best of my memory.

> (is all damage supposed to shear?)

Not sure. I'll check.

**Comment by exaskye** [comment]
Yes, all damage shears.

https://pool.jortage.com/sleepingtown/media_attachments/files/105/797/158/354/781/662/original/529620fb1c1e2714.mp4

(Wow, that video is laggy.)

File: src/main/java/com/unascribed/fabrication/mixin/i_woina/old_sheep_shear/MixinSheepEntity.java (hunk 2 above)
Comment (SFort) [reply src/main/java/com/unascribed/fabrication/mixin/i_woina/old_sheep_shear/MixinSheepEntity.java]: Was generated when extending AnimalEntity for some reason(will remove)

**Comment by SFort** [comment]
went and checked sheep did in fact have to be dammaged via player

**Review by exaskye — COMMENTED** [review]
I've still got some issues with this but I'll fix it myself.

**Pull request merged**